src/geometry/tri_mesh.rs

```diff
@@ -446,6 +446,103 @@ impl TriMesh {
         arrow
     }
 
+    ///
+    /// Returns a ribbon mesh from the passed parameter "paths"
+    /// Paths is a vector of paths, where each path is a vector of Vec3.
+    /// Each path is a list of successive points. Paths will be connected with triangles to make a surface or a volume.
+    /// All the paths are required to have the same number of points.
+    /// At least two paths are required to create a ribbon.
+    /// At least each path should have two points.
+    /// 
+    pub fn ribbon(paths: &Vec<Vec<Vec3>>) -> Self {
+        // path lengths
+        let p = paths.len();
+        if p < 2 {
+            panic!("At least two paths are required to create a ribbon");
```

Comment:
The idea of Rust is to catch errors at compile-time, so if it's at all possible, the checks should be compile-time instead of runtime as you are doing here. It's of course not always possible to do compile-time checks, and I'm not sure it's possible in this case, just wanted to mention it.